### PR TITLE
fix(NODE-2899): sort and correct circular imports

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

- A number of constants are moved to a shared leaf module.
- moved `commandSupportsReadConcern` to utils.ts for sharing
- Auth providers map can be defined inside the only module that uses it
- changed the defaultAuthProviders file name, it was the only camelcase one
- sorted all the imports and added that to our linting
- added testing to make sure each file can be imported on its own

#### What is the motivation for this change?

Imports are sorted, and we are checking to be sure circular dependency issues aren't reintroduced.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- New TODOs have a related JIRA ticket

Flakey test discovered: https://jira.mongodb.org/browse/NODE-3808

### CODE REVIEW

File: src/cmap/connection.ts
Comment (durran) [inline src/cmap/connection.ts:200]: Just a question here. We are importing constants here and then redefining them as static variables on this class. Why not just use the original constants everywhere?

File: src/cmap/connection_pool.ts
Comment (durran) [inline src/cmap/connection_pool.ts:148]: Same as my previous comment on the constants.

File: src/sdam/server.ts
Comment (durran) [inline src/sdam/server.ts:125]: Same on the constants front here. Maybe this is a new ticket to clean this up and import directly everywhere?

**Review by durran — DISMISSED** [review]

**Comment by dariakp** [comment]
Note: Marking blocked on the merge of our two outstanding feature PRs

**Comment by dariakp** [comment]
Removing blocked label since the feature PR went in (and we shouldn't need to worry about this for the dot notation one); just needs a rebase before re-review and merge

**Review by dariakp — APPROVED** [review]
LGTM, tyvm

**Pull request merged**
